cmd/sealer/cmd/image/build.go

```diff
@@ -160,13 +161,15 @@ func buildSealerImage() error {
 	}
 
 	defer func() {
-		for _, m := range []string{tempTag} {
-			// the above image is intermediate image, we need to remove it when the build ends.
-			if err := engine.RemoveImage(&bc.RemoveImageOptions{
-				ImageNamesOrIDs: []string{m},
-				Force:           true,
-			}); err != nil {
-				logrus.Debugf("failed to remove image %s, you need to remove it manually: %v", m, err)
+		if buildFlags.DownloadContainerImage {
```

Comment:
we could skip download the container image if it is false, we do not need to do `applyRegistryToImage`